packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -72,266 +79,136 @@ class _DeepLinkListViewMainPanel extends StatelessWidget {
-    required this.tableView,
-    required this.controller,
-    required this.showSpitScreen,
-  });
-  final List<LinkData> linkDatas;
-  final TableViewType tableView;
-  final DeepLinksController controller;
-  final bool showSpitScreen;
-
-  @override
-  Widget build(BuildContext context) {
-    return Row(
-      children: [
-        Expanded(
-          child: _DataTable(
-            tableView: tableView,
-            linkDatas: linkDatas,
-            controller: controller,
-          ),
-        ),
-        if (showSpitScreen)
-          Expanded(
-            child: ValueListenableBuilder<LinkData?>(
-              valueListenable: controller.selectedLink,
-              builder: (context, selectedLink, _) => _ValidationDetailScreen(
-                tableView: tableView,
-                linkData: selectedLink!,
+              ],
+            );
+          }
+          return Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              _NotificationCardSection(
+                domainErrorCount: displayOptions.domainErrorCount,
+                pathErrorCount: displayOptions.pathErrorCount,
                 controller: controller,
               ),
-            ),
-          ),
-      ],
+              Expanded(
+                child: _AllDeepLinkDataTable(controller: controller),
+              ),
+            ],
+          );
+        },
+      ),
     );
   }
 }
 
 class _DataTable extends StatelessWidget {
   const _DataTable({
     required this.linkDatas,
-    required this.tableView,
+    required this.viewType,
     required this.controller,
   });
   final List<LinkData> linkDatas;
-  final TableViewType tableView;
+  final TableViewType viewType;
   final DeepLinksController controller;
 
   @override
   Widget build(BuildContext context) {
-    final ColumnData<LinkData> domain = DomainColumn();
-    final ColumnData<LinkData> path = PathColumn();
-
-    return FlatTable(
-      keyFactory: (node) => ValueKey(node.toString),
-      data: linkDatas,
-      dataKey: 'deep-links',
-      autoScrollContent: true,
-      columns: <ColumnData>[
-        if (tableView != TableViewType.pathView) domain,
-        if (tableView != TableViewType.domainView) path,
-        SchemeColumn(),
-        OSColumn(),
-        if (!controller.showSpitScreen) ...[
-          StatusColumn(),
-          NavigationColumn(),
-        ],
-      ],
-      selectionNotifier: controller.selectedLink,
-      defaultSortColumn: tableView == TableViewType.pathView ? path : domain,
-      defaultSortDirection: SortDirection.ascending,
-      onItemSelected: (item) => controller.showSpitScreenNotifier.value = true,
-    );
-  }
-}
-
-class _ValidationDetailScreen extends StatelessWidget {
-  const _ValidationDetailScreen({
-    required this.linkData,
-    required this.tableView,
-    required this.controller,
-  });
+    final ColumnData<LinkData> domain = DomainColumn(controller);
```

Comment:
nit: remove unnecessary types on LHS for `domain` and `path`

Reply:
if remove types here, `domain` and `path` will be recognized as Object.
 And will need to cast them to `ColumnData` in `defaultSortColumn` and `columns`

Reply:
I still need to cast them to ColumnData<LinkData> in defaultSortColumn and columns.

It's because i used `switch` in columns and '?:' in defaultSortColumn so they are not very smart.

So I removed the LHS types and added the cast